### PR TITLE
[Basic] Keep SwiftPM compatible with Swift 4.0

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
Depends on https://github.com/apple/swift-package-manager/pull/1521

File: Sources/Basic/misc.swift [hunk 1]
```diff
@@ -95,3 +95,11 @@ public func ==<E: Equatable>(a: [[E]], b: [[E]]) -> Bool {
     }
     return !zip(a, b).contains { $0 != $1 }
 }
+
+#if !swift(>=4.1)
```
Comment (aciidb0mb3r) [inline Sources/Basic/misc.swift]: @jrose-apple is this fine?

Comment (jrose-apple) [reply Sources/Basic/misc.swift]: Yep, that should work okay.

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Pull request merged**